docs/manual/scala/guide/cluster/PersistentEntity.md

```diff
@@ -0,0 +1,221 @@
+
+The events must be immutable to avoid concurrency issues that may occur from changing an event instance that is about to be persisted.
+
+The section [[Immutable Objects|Immutable]] describes how to define immutable event classes.
+
+## Replies
+
+Each command must define what type of message to use as reply to the command by implementing the [PersistentEntity.ReplyType](api/index.html?com/lightbend/lagom/scaladsl/persistence/PersistentEntity.ReplyType.html) interface.
+
+@[AddPost](code/docs/home/scaladsl/persistence/BlogCommand.scala)
+
+You send the reply message using the `reply` method of the context that is passed to the command handler function. Note that the reply message type must match the `ReplyType` defined by the command, and by the second type parameter of `onCommand`.
+
+Typically the reply will be an acknowledgment that the entity processed the command successfully, i.e. you send it after persist.
+
+@[reply](code/docs/home/scaladsl/persistence/Post2.scala)
+
+For convenience you may use the `akka.Done` as acknowledgment message.
+
+It can also be a reply to a read-only query command.
+
+@[read-only-command-handler](code/docs/home/scaladsl/persistence/Post2.scala)
+
+You can use `ctx.invalidCommand` to reject an invalid command, which will fail the `Future` with `PersistentEntity.InvalidCommandException` on the sender side.
+
+You can send a negative acknowledgment with `ctx.commandFailed`, which will fail the `Future` on the sender side with the given exception.
+
+If persisting the events fails a negative acknowledgment is automatically sent, which will fail the `Future` on the sender side with `PersistentEntity.PersistException`.
+
+If the `PersistentEntity` receives a command for which there is no registered command handler a negative acknowledgment is automatically sent, which will fail the the `Future` on the sender side with `PersistentEntity.UnhandledCommandException`.
+
+If you don't reply to a command the `Future` on the sender side will be completed with a `akka.pattern.AskTimeoutException` after a timeout.
+
+## Changing Behavior
+
+For simple entities you can use the same set of command and event handlers independent of what state the entity is in. The actions can then be defined like this:
+
+@[same-behavior](code/docs/home/scaladsl/persistence/Post3.scala)
+
+When the state changes it can also change the behavior of the entity in the sense that new functions for processing commands and events may be defined. This is useful when implementing finite state machine (FSM) like entities. The `Actions`, the set of event handler and command handlers, can be selected based on current state. The return type of the `behavior` method is a function from current `State` to `Actions`. The reason `Actions` can be used as in the above example is because `Actions` itself is such a function returning itself.
```

Comment:
`Action itself is such a function returning itself` -> `Actions is such a function returning itself for any State` ?